main/src/library/Fixpoint/Solver.flix

```diff
@@ -436,8 +436,15 @@ namespace Fixpoint {
     ///
     def factsOf(f: Array[HeadTerm[v]] -> t & e, p: PredSym, d: Datalog[v]): Array[t] & Impure =
         let Datalog(facts, _) = d;
-        restrict(facts, Set#{p}) |>
-        Array.map(match Constraint(HeadAtom(_, _, terms), _) -> f(terms))
+        let factsP = MutList.new();
+        MutList.reserve!(Array.length(facts), factsP);
```

Comment:
Is this a problem if `facts` is huge and `p` is comparatively small?

Reply:
What is the problem?

Reply:
Oh I get it. 

Yes ,its a problem. In general we should Datalog to "compute something big" but the result will often be small.